### ISSUE TITLE
Package native wallet libraries in universal jar file [2 XMR]

### ISSUE THREAD

**Issue by woodser** [haveno-dex/haveno#844]
This issue requests packaging the native wallet libraries for monero-java in a universal jar file for easy distribution.

This issue shares a bounty with the related issue in monero-java: https://github.com/woodser/monero-java/issues/12

**Comment by niyid** [comment]
@woodser

Can I have a go at this? Assign please.🙏🏿 Thanks.

**Comment by woodser** [comment]
Thanks @niyid. I've assigned to you. If needed after some time, we can open it back up for others to help, depending on progress.

Thanks again.

**Comment by woodser** [comment]
Please use [lmdbjava](https://github.com/lmdbjava/lmdbjava) and [native platform](https://github.com/adammurdoch/native-platform) for reference.

**Comment by niyid** [comment]
@woodser 

Please, why would I get this when I try to make the latest Haveno:

![Screenshot from 2024-03-27 22-47-48](https://github.com/haveno-dex/haveno/assets/20237127/09faef40-cf3a-4465-913d-a9a88c5605dc)

**Comment by woodser** [comment]
You need to update your JDK to 21:

```
curl -s "https://get.sdkman.io" | bash
sdk install java 21.0.2.fx-librca
```

**Comment by niyid** [comment]
> You need to update your JDK to 21:
> 
> ```
> curl -s "https://get.sdkman.io" | bash
> sdk install java 21.0.2.fx-librca
> ```

Thanks

**Comment by niyid** [comment]
After building using make, I still do not see the expected downloaded libraries as described in https://github.com/haveno-dex/haveno/issues/783

**Comment by woodser** [comment]
That issue isn't relevant anymore.

There's a PR open to download and run the native libraries in Haveno: https://github.com/haveno-dex/haveno/pull/813

But please note, this work should happen in the monero-java library, to package the native libraries into a jar for general distribution, which Haveno can also consume.

**Comment by niyid** [comment]
Ah, OK. I'm looking at using Gradle Shadow Plugin. I hope you find that
agreeable.

**Comment by woodser** [comment]
If it works, but that project's already using Maven. I'm open to using Gradle, but at least they should not conflict.

**Comment by niyid** [comment]
I suppose the native libraries are prebuilt? Where can I download them?

**Comment by woodser** [comment]
https://github.com/woodser/monero-java/releases/tag/v0.8.13

**Comment by niyid** [comment]
Is this what you need?

https://drive.google.com/file/d/1zg7YicKOXfgRdeuEYOfCmxEfIfaYpSrG/view?usp=drive_link

**Comment by woodser** [comment]
I'm expecting a PR to monero-java to create the bundled jar on my own.

**Comment by niyid** [comment]
Yes. I will follow up with that. I just want to confirm if that is the
bundled jar you want.

**Comment by woodser** [comment]
This issue is completed with this commit to monero-java: https://github.com/woodser/monero-java/commit/65d3aee03e2854405aaa66e2b42ae4848bca332b

Please post or DM an address for payment. Thanks!

**Comment by niyid** [comment]
88Yz73BwJRU4cxqqfaWdincrm2naHcTrvMPzSMwyAZU5XhYm6HYUqhX1nPEtkhsgEVAKFjTTj7wAURWoWbAZsvbu1WPaG1j

Thanks a lot🙏🏿

**Comment by niyid** [comment]
@woodser

This is weird. My wallet address is below:

43fLNsTYYbeYFSg1rqSb5mRfj7Cdq2ZLNAvWUiDguZA3Xuxkh9RWc2mggcGSnvs7cGAHvi7fEm1PicKAh6YQ6DWTQrQvns3

**Comment by niyid** [comment]
> Payment sent. Thanks!

Received. Thanks. Apologies for the hassle.
